### PR TITLE
Node structure tests

### PR DESCRIPTION
**Pull request by aaron-prindle**

### CODE REVIEW

File: reproducible/ubuntu/node_gen_tar.bzl [hunk 1]
```diff
@@ -73,17 +73,23 @@ load(
...
 )
 
-def nodebootstrap_image(name, env=None):
+def node_gen_tar_rule(name, env=None):
+    docker_build(
+        name = "%s_builder" % name,
+        base = ":ubuntu_build",
+        entrypoint = [
+            "/mknodeimage.sh",
+            "%s" % name[len("node_"):].replace("_", "."),
```
Comment (nkubala) [inline reproducible/ubuntu/node_gen_tar.bzl]: nit: can you just remove the `"%s" %` here?

File: tests/node_8_5_0_ubuntu_build_test.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  args: ['--version']
+  expectedOutput: ['v8.5.0']
+metadataTest:
+  env:
+    - key : PORT
+      value: 8080
+    - key: NODE_ENV
+      value: production
+  exposedPorts: ["8080"]
+#  entrypoint: []
```
Comment (nkubala) [inline tests/node_8_5_0_ubuntu_build_test.yaml]: remove

File: tests/node_8_5_0_ubuntu_test.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  args: ['--version']
+  expectedOutput: ['v8.5.0']
+metadataTest:
+  env:
+    - key : PORT
+      value: 8080
+    - key: NODE_ENV
+      value: production
+  exposedPorts: ["8080"]
+#  entrypoint: []
```
Comment (nkubala) [inline tests/node_8_5_0_ubuntu_test.yaml]: remove

File: tests/ubuntu_build_test.yaml [hunk 4]
```diff
@@ -1,15 +1,11 @@
...
-  path: '/etc/os-release'
-fileExistenceTests:
-- name: 'Root'
-  path: '/'
-  shouldExist: true
-  permissions: 'drwxr-xr-x'
+- name: 'python on path'
+  command: 'which'
+  args: ['python']
+  expectedOutput: ['/usr/bin/python2.7']
```
Comment (nkubala) [inline tests/ubuntu_build_test.yaml]: not totally sure but I think you need a newline here? these match regexes so you would need either newline or `.*` at the end

File: tests/node_8_5_0_ubuntu_build_test.yaml (hunk 2 above)
Comment (aaron-prindle) [reply tests/node_8_5_0_ubuntu_build_test.yaml]: I put this here a a stub as I believe we should test the entrypoint but I wasn't sure what it should be.  Will change to the appropriate value.

File: reproducible/ubuntu/node_gen_tar.bzl (hunk 1 above)
Comment (aaron-prindle) [reply reproducible/ubuntu/node_gen_tar.bzl]: I did this originally but had an error.  It doesn't make sense that it's related though, will change back.

File: tests/ubuntu_build_test.yaml (hunk 4 above)
Comment (aaron-prindle) [reply tests/ubuntu_build_test.yaml]: These are passing locally as is.  The "echo $PATH" test that i used as a template doesn't append a newline even though the output has it.  Going  to leave as is as it appears to be working and is a little cleaner.

File: tests/node_8_5_0_ubuntu_build_test.yaml (hunk 2 above)
Comment (aaron-prindle) [reply tests/node_8_5_0_ubuntu_build_test.yaml]: going to drop this actually, ubuntu is the only image that has a defined entrypoint, the rest inherit it

File: tests/ubuntu_build_test.yaml (hunk 4 above)
Comment (nkubala) [reply tests/ubuntu_build_test.yaml]: ok yeah sounds good, if it's working don't change it just thought it might fail

**Comment by nkubala** [comment]
looks like a build error, but the logs are exploding with 
```
[452 / 466] 2 actions running
 SkylarkAction reproducible/ubuntu/node_8_6_0.tar.gz; 2010s linux-sandbox
 SkylarkAction reproducible/ubuntu/node_8_5_0.tar.gz; 1928s linux-sandbox
```

any idea?

**Comment by aaron-prindle** [comment]
yea, this is going in after https://github.com/GoogleCloudPlatform/debian-docker/pull/112.  Put this here to get review in parallel.  Currently this is timing out as the images take a long time to build.  Currently working on speeding builds up w/ some bazel optimizations and caching in Travis: https://github.com/GoogleCloudPlatform/debian-docker/pull/114

**Pull request merged**